src/control/cmd/dmg/system.go

```diff
@@ -414,3 +418,175 @@ func (cmd *systemDelAttrCmd) Execute(_ []string) error {
+		} else {
+			return errors.Errorf("invalid system property key: %s", k)
+		}
+	}
+
+	return nil
+}
+
+func (f *systemSetPropsFlag) Complete(match string) []flags.Completion {
+	if f.SystemProps == nil {
+		f.SystemProps = daos.SystemProperties()
+	}
+
+	comps := make(ui.CompletionMap)
+	for _, prop := range f.SystemProps {
+		comps[prop.Key.String()] = prop.Value.Choices()
+	}
+	f.SetCompletions(comps)
+
+	return f.SetPropertiesFlag.Complete(match)
+}
+
+// systemSetPropCmd represents the command to set system properties.
+type systemSetPropCmd struct {
+	baseCmd
+	cfgCmd
+	ctlInvokerCmd
+	jsonOutputCmd
+
+	Args struct {
+		Props systemSetPropsFlag `positional-arg-name:"system properties to set (key:val[,key:val...])" required:"1"`
+	} `positional-args:"yes"`
+}
+
+// Execute is run when systemSetPropCmd subcommand is activated.
+func (cmd *systemSetPropCmd) Execute(_ []string) error {
+	req := &control.SystemSetPropReq{
+		Properties: cmd.Args.Props.ParsedProps,
+	}
+
+	err := control.SystemSetProp(context.Background(), cmd.ctlInvoker, req)
+	if cmd.jsonOutputEnabled() {
+		return cmd.outputJSON(nil, err)
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "system set-prop failed")
+	}
+	cmd.Info("system set-prop succeeded")
+
+	return nil
+}
+
+type systemGetPropsFlag struct {
+	ui.GetPropertiesFlag
+	SystemProps daos.SystemPropertyMap
+	ParsedProps []daos.SystemPropertyKey
+}
+
+func (f *systemGetPropsFlag) UnmarshalFlag(fv string) error {
+	err := f.GetPropertiesFlag.UnmarshalFlag(fv)
+	if err != nil {
+		return err
+	}
+
+	if f.SystemProps == nil {
+		f.SystemProps = daos.SystemProperties()
+	}
+
+	for _, k := range f.GetPropertiesFlag.ParsedProps.ToSlice() {
+		if prop, ok := f.SystemProps.Get(k); ok {
+			f.ParsedProps = append(f.ParsedProps, prop.Key)
+		} else {
+			return errors.Errorf("invalid system property key: %s", k)
+		}
+	}
+
+	return nil
+}
+
+func (f *systemGetPropsFlag) Complete(match string) []flags.Completion {
+	if f.SystemProps == nil {
+		f.SystemProps = daos.SystemProperties()
+	}
+
+	comps := make(ui.CompletionMap)
+	for _, prop := range f.SystemProps {
+		comps[prop.Key.String()] = prop.Value.Choices()
+	}
+	f.SetCompletions(comps)
+
+	return f.GetPropertiesFlag.Complete(match)
+}
+
+// systemGetPropCmd represents the command to get system properties.
+type systemGetPropCmd struct {
+	baseCmd
+	cfgCmd
+	ctlInvokerCmd
+	jsonOutputCmd
+
+	Args struct {
+		Props systemGetPropsFlag `positional-arg-name:"system properties to get (key[,key...])"`
+	} `positional-args:"yes"`
+}
+
+func prettyPrintSysProps(out io.Writer, props []*daos.SystemProperty) {
+	if len(props) == 0 {
+		fmt.Fprintln(out, "No system properties found.")
+		return
+	}
+
+	nameTitle := "Name"
+	valueTitle := "Value"
+	table := []txtfmt.TableRow{}
+	for _, prop := range props {
+		row := txtfmt.TableRow{}
+		row[nameTitle] = fmt.Sprintf("%s (%s)", prop.Description, prop.Key)
+		row[valueTitle] = prop.Value.String()
```

Comment:
I might be inclined to split the description from the key, for clarity.

Reply:
For the moment, anyhow, I wanted to be consistent with the [output](https://github.com/daos-stack/daos/blob/5b00785411d972dd001626069aa2c9234592b2a1/src/control/cmd/dmg/pretty/pool.go#L304) of `dmg pool get-prop`. At some point, it would be a good idea to refactor a bit so that we have shared code for printing all properties, but I didn't want to tackle that as part of this work.

Reply:
Makes sense.